diff --git a/OneWire.cpp b/OneWire.cpp
index cb0d670..edce492 100644
--- a/OneWire.cpp
+++ b/OneWire.cpp
@@ -596,8 +596,8 @@ uint16_t OneWire::crc16(const uint8_t* input, uint16_t len, uint16_t crc)
 
 // undef defines for no particular reason
 #ifdef ARDUINO_ARCH_ESP32
-#  undef noInterrupts() {portMUX_TYPE mux = portMUX_INITIALIZER_UNLOCKED;portENTER_CRITICAL(&mux)
-#  undef interrupts() portEXIT_CRITICAL(&mux);}
+#  undef noInterrupts
+#  undef interrupts
 #endif
 // for info on this, search "IRAM_ATTR" at https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/general-notes.html 
 #undef CRIT_TIMING 
diff --git a/OneWire.h b/OneWire.h
index a7bfab7..bc6280b 100644
--- a/OneWire.h
+++ b/OneWire.h
@@ -1,6 +1,14 @@
 #ifndef OneWire_h
 #define OneWire_h
 
+#if defined(ARCH_ESP32)
+#define ARDUINO_ARCH_ESP32
+#elif defined(ARCH_ESP8266)
+#define ARDUINO_ARCH_ESP8266
+#elif defined(ARCH_RP2040)
+#define ARDUINO_ARCH_RP2040
+#endif
+
 #ifdef __cplusplus
 
 #include <stdint.h>
@@ -71,6 +79,7 @@ class OneWire
   public:
     OneWire() { }
     OneWire(uint8_t pin) { begin(pin); }
+    void begin() { } // Do nothing: constructor did actual initialisation
     void begin(uint8_t pin);
 
     // Perform a 1-Wire reset cycle. Returns 1 if a device responds
diff --git a/util/OneWire_direct_gpio.h b/util/OneWire_direct_gpio.h
index a63aae1..c527914 100644
--- a/util/OneWire_direct_gpio.h
+++ b/util/OneWire_direct_gpio.h
@@ -164,6 +164,8 @@ bool directRead(IO_REG_TYPE mask)
 #define IO_REG_BASE_ATTR
 #define IO_REG_MASK_ATTR
 
+#define digitalPinIsValid(pin)   GPIO_IS_VALID_GPIO(pin)
+
 static inline __attribute__((always_inline))
 IO_REG_TYPE directRead(IO_REG_TYPE pin)
 {
